### PR TITLE
Fixing compability error

### PR DESCRIPTION
**Pull request by cnohall**
This is related to issue #99. It will work as it should if we add ` Context::getContext()->currency = Context::getContext()->currency ?? new Currency(Configuration::get('PS_CURRENCY_DEFAULT'));` on the line before `$o->setCurrentState(Configuration::get('PS_OS_PAYMENT'));`

### CODE REVIEW

**Comment by DarrenWestwood** [comment]
Ensure callbacks credit the correct amount to the order if a user has selected a different currency to the default currency:
https://zemez.io/prestashop/support/how-to/prestashop-1-7-manage-currencies-set-default-one/

**Comment by cnohall** [comment]
I've updated so that the currency_id is fetched from the cart instead of going for the shop's default currency. As a fallback the currency is fetched from the user's cache.

**Comment by DarrenWestwood** [comment]
`Context::getContext()->cart->id_currency` or `new Currency($cookie->id_currency);` will not work in this case, as callbacks are delivered from Blockonomics and the user cart and cookie will not be available. We should use the currency from the blockonomics_order or the PrestaShop order after the order has been located from the address.

**Comment by cnohall** [comment]
Thanks, makes a lot of sense. I've changed it accordingly and pushed.

File: callback.php [hunk 1]
```diff
@@ -89,6 +89,14 @@
                 if ($order[0]['bits'] > $order[0]['bits_payed']) {
                     $o->setCurrentState(Configuration::get('PS_OS_ERROR'));
                 } else {
+                    $id_currency = Db::getInstance()->ExecuteS(
```
Comment (DarrenWestwood) [inline callback.php:92]: We have already collected the id_currency along with the PrestaShop order $o. Simply using `$o->id_currency` should produce the same value as collecting it from the database again `$id_currency[0]["id_currency"]`

Comment (cnohall) [reply callback.php]: You are right, I changed it and tried it. Works excellent. Thanks!

**Review by DarrenWestwood — APPROVED** [review]
The fix is neat. It fixes the issue in versions > 1.7.7.2 and tested it does not affect previous versions.

**Pull request merged**